src/nimony/sem.nim

```diff
@@ -222,15 +222,28 @@ proc semInclude(c: var SemContext; it: var Item) =
 
   producesVoid c, info, it.typ
 
-proc importSingleFile(c: var SemContext; f1, origin: string; info: PackedLineInfo) =
-  let f2 = resolveFile(c, origin, f1)
+proc importSingleFile(c: var SemContext; f1: ImportedFilename; origin: string; info: PackedLineInfo) =
+  let f2 = resolveFile(c, origin, f1.path)
   let suffix = moduleSuffix(f2, c.g.config.paths)
   if not c.processedModules.containsOrIncl(suffix):
     c.meta.importedFiles.add f2
     if needsRecompile(f2, suffix):
       selfExec c, f2
 
-    loadInterface suffix, c.importTab
+    let moduleName = pool.strings.getOrIncl(f1.name)
+    let moduleSym = identToSym(c, moduleName, ModuleY)
+    let s = Sym(kind: ModuleY, name: moduleSym, pos: ImportedPos)
+    c.currentScope.addOverloadable(moduleName, s)
+    var moduleDecl = createTokenBuf(2)
+    moduleDecl.addParLe(ModuleY, info)
+    moduleDecl.addParRi()
+    publish moduleSym, moduleDecl
+    var module = ImportedModule()
+    loadInterface suffix, module.iface
+    for strId, _ in module.iface:
+      # mark that this module contains the identifier `strId`:
+      c.importTab.mgetOrPut(strId, @[]).add(moduleSym)
```

Comment:
I think we can make `loadInterface` do this step too and thus avoid another iteration over the iface (which could be large).